platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/SyncGossip.java

```diff
@@ -174,7 +179,20 @@ public SyncGossip(
         final ThreadConfig threadConfig = platformContext.getConfiguration().getConfigData(ThreadConfig.class);
 
         final BasicConfig basicConfig = platformContext.getConfiguration().getConfigData(BasicConfig.class);
-        final List<PeerInfo> peers = Utilities.createPeerInfoList(roster, selfId);
+        final RosterEntry selfEntry = RosterUtils.getRosterEntry(roster, selfId.id());
+        final List<PeerInfo> peers;
+        if (RosterUtils.fetchGossipCaCertificate(selfEntry) == null) {
```

Comment:
why do we even make it this far if we dont have a valid cert? why not die immediately

Reply:
In this implementation the platform commits sepuku in the platform builder when it discovers the cert in `KeysAndCerts` is bad.